app/src/main/java/com/github/libretube/ui/dialogs/DownloadDialog.kt

```diff
@@ -32,10 +32,17 @@ import kotlinx.coroutines.launch
 import kotlinx.coroutines.withContext
 import retrofit2.HttpException
 
-class DownloadDialog(
-    private val videoId: String
-) : DialogFragment() {
+class DownloadDialog : DialogFragment() {
+    private lateinit var videoId: String
     private var onDownloadConfirm = {}
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        arguments?.let {
```

Comment:
```suggestion
        videoId = arguments.getString(IntentData.videoId)
```

Reply:
It must be like this: `videoId = arguments?.getString(IntentData.videoId)!!`

Reply:
True